### PR TITLE
adapter,controller: add replica metrics history

### PR DESCRIPTION
**Pull request by teskje**
This PR adds a new builtin source `mz_cluster_replica_metrics_history`, and a corresponding view `mz_cluster_replica_utilization_history` on top of it. They contain the data already present in `mz_cluster_replica_{metrics,utilization}`, but as (pseudo-) append-only collections, rather than retained-metrics collections.

This plan is to remove the old retained-metrics collections in favor of the new append-only ones, but we need to wait until the new ones had time to backfill the historical data present in the old ones.

### Motivation

  * This PR adds a known-desirable feature.

Part of https://github.com/MaterializeInc/materialize/issues/28795

### Tips for reviewer

Some of our user docs (Troubleshooting, Grafana, Datadog) refer to `mz_cluster_replica_utilization`. I'm not touching these here, but we'll need to change them when/before we remove that relation.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: src/storage-client/src/healthcheck.rs [hunk 1]
```diff
@@ -138,3 +138,17 @@ pub static MZ_AWS_PRIVATELINK_CONNECTION_STATUS_HISTORY_DESC: LazyLock<RelationD
...
+        .with_column("process_id", ScalarType::UInt64.nullable(false))
+        .with_column("cpu_nano_cores", ScalarType::UInt64.nullable(true))
+        .with_column("memory_bytes", ScalarType::UInt64.nullable(true))
+        .with_column("disk_bytes", ScalarType::UInt64.nullable(true))
+        .with_column(
+            "occurred_at",
+            ScalarType::TimestampTz { precision: None }.nullable(false),
+        )
+        .finish()
+});
```
Comment (teskje) [inline src/storage-client/src/healthcheck.rs:154]: This doesn't really fit into `healthcheck.rs`, but we already have the statement execution history stuff here.

**Review by jkosh44 — APPROVED** [review]
Adapter changes LGTM

**Review by aljoscha — APPROVED** [review]
I only looked at the new truncation method, and that one looked good to me!

**Comment by teskje** [comment]
TFTRs!

**Pull request merged**